core/src/labels/labels.cpp

```diff
@@ -314,14 +316,12 @@ void Labels::update(const View& _view, float _dt,
         if (label->options().repeatDistance == 0.f) {
             continue;
         }
-        TextLabel* textLabel = dynamic_cast<TextLabel*>(label);
-        if (!textLabel) { continue; }
-        repeatGroupSet.push_back(textLabel);
+        repeatGroupSet.push_back(label);
```

Comment:
It does only add TextLabel. Until we modify PointStyle to set repeatDistance there is no chance that anything but a TextLabel passes ```label->options().repeatDistance != 0.f```. The question is whether to make this explicit by adding a dynamic cast like:
```
        if (label->options().repeatDistance == 0.f ||
            !dynamic_cast<TextLabel*>(label)) {
            continue;
        }

```